src/service/video-manager-impl.js

```diff
@@ -437,6 +556,216 @@ class VideoEntry {
   }
 
   /**
+   * Called when the video's position in the viewport changed (at most once per
+   * animation frame)
+   * @param {PositionInViewportEntryDef} newPos
+   * @private
+   */
+  onDockableVideoPositionChanged_(newPos) {
+
+
+    const inViewportHeight = this.calculateDockableVideoPosition_(newPos);
+
+    // Short-circuit the position change handler if the video isn't loaded yet
+    // or is not playing manually while in-line (paused videos need to go
+    // through if they are docked since this method handles the "undocking"
+    // animation)
+    if (!this.loaded_
+      || !this.internalElement_
+      || (
+        this.getPlayingState() != PlayingStates.PLAYING_MANUAL
+        && !this.internalElement_.classList.contains(DOCK_CLASS)
+      )
+    ) {
+      return;
+    }
+
+    // Temporary fix until PositionObserver somehow tracks objects outside of
+    // the viewport
+    if (inViewportHeight == 0
+       && this.getPlayingState() == PlayingStates.PLAYING_MANUAL
+       && this.minimizePosition_ != MinimizePositions.DEFAULT) {
+      this.endDocking_();
+      return;
+    }
+
+    if (
+      // Minimize the video if it's out of view and it has been viewed
+      // at least once before
+      this.minimizePosition_ != MinimizePositions.INVIEW
```

Comment:
please breakdown into multiple boolean consts for better readability of this condition.